### ISSUE TITLE
pdk validate fails if host puppet.conf contains deprecated settings

### ISSUE THREAD

**Issue by jdowling** [puppetlabs/pdk#304]
Causes jenkins jobs to go unstable as this is logged as test failures when published as junit test results.

This should not happen as the contents of the module under test shouldn't have anything to do with the host configuration. It would make more sense for pdk to contain a stub puppet.conf & override reference to that when puppet is invoked.

Repro scenario:
host happens to be managed by an older version of puppet.
The system puppet.conf on the host contains 'configtimeout'
command to run: `pdk validate --format=junit:validate.xml   && exit %ERRORLEVEL%`

output:
```
...
<testsuite name="puppet-syntax" tests="3" errors="0" failures="2" skipped="0" time="0" timestamp="2017-09-29T15:49:37" hostname="hosty" id="2" package="puppet-syntax">
<properties/>
<testcase classname="puppet-syntax" name="manifests/init.pp" time="0"/>
<testcase classname="puppet-syntax" name="" time="0">
<failure type="Warning" message="Setting configtimeout is deprecated.">
Warning: puppet-syntax: Setting configtimeout is deprecated.
</failure>
</testcase>
<testcase classname="puppet-syntax" name="" time="0">
<failure type="" message="(at C:/Program Files/Puppet Labs/DevelopmentKit/share/cache/ruby/2.1.0/gems/puppet-5.2.0-x64-mingw32/lib/puppet/settings.rb:1159:in `issue_deprecation_warning')">
: puppet-syntax: (at C:/Program Files/Puppet Labs/DevelopmentKit/share/cache/ruby/2.1.0/gems/puppet-5.2.0-x64-mingw32/lib/puppet/settings.rb:1159:in `issue_deprecation_warning')
</failure>
</testcase>
<system-out/>
<system-err/>
</testsuite>
...
```

**Comment by DavidS** [comment]
I could reproduce this locally, and have created https://tickets.puppetlabs.com/browse/PDK-575 for tracking this internally.

While I consider this a bug, I would also recommend against running testing and validation on a privileged account, to avoid unfortunate "accidents" with misbehaving tests.

**Comment by scotje** [comment]
This issue should now be resolved as of PDK 1.4.1, please let us know if you are still having difficulties after upgrading. Thanks for the bug report!
